src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java

```diff
@@ -666,4 +676,85 @@ public static boolean supportJDBC43(Connection con) throws SQLException {
     public static String escapeSingleQuotes(String name) {
         return name.replace("'", "''");
     }
-}
+    
+    /**
+     * Returns if connected to SQL Azure
+     * 
+     * @param con
+     *        connection to server
+     * @return boolean
+     * @throws SQLException
+     */
+    public static boolean isSqlAzure(Connection con) throws SQLException {
+        if (_determinedSqlAzureOrSqlServer) {
+            return _isSqlAzure;
+        }
+
+        try {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+        } catch (ClassNotFoundException e) {
+            throw new SQLException(e.toString());
+        }
+
+        boolean ownsCon = false;
+
+        if (null == con) {
+            con = DriverManager.getConnection(AbstractTest.getConnectionString());
+            ownsCon = true;
+        }
+
+        try (Statement stmt = con.createStatement();
+                ResultSet rs = stmt.executeQuery("SELECT CAST(SERVERPROPERTY('EngineEdition') as INT)")) {
+            rs.next();
+            int engineEdition = rs.getInt(1);
+            _determinedSqlAzureOrSqlServer = true;
+            _isSqlAzure = (engineEdition == ENGINE_EDITION_FOR_SQL_AZURE
+                    || engineEdition == ENGINE_EDITION_FOR_SQL_AZURE_DW);
+            if (ownsCon) {
+                con.close();
+            }
+            return _isSqlAzure;
+        }
+    }
+    
+    /**
+     * Returns if connected to SQL Azure DW
+     * 
+     * @param con
+     *        connection to server
+     * @return boolean
+     * @throws SQLException
+     */
+    public static boolean isSqlAzureDW(Connection con) throws SQLException {
+        if (_determinedSqlAzureOrSqlServer) {
+            return _isSqlAzureDW;
+        }
+
+        try {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
```

Comment:
This may be unnecessary block

Reply:
This is necessary because this static block can be called in the JUnit test before the classloader has had a chance to load our driver. The JUnit test fails without this line here (because sometimes we want to identify the server before we even use the driver).